test/unit/deadline_client/cli/test_cli_asset.py

```diff
@@ -0,0 +1,171 @@
+
+    def test_valid_manifest_out(self, tmp_path):
+        """
+        Tests if CLI snapshot command correctly calls with --manifest-out arguement
+        """
+        root_dir = str(tmp_path)
+        manifest_out_dir = tmp_path / "manifest_out"
+        manifest_out_dir.mkdir()
+
+        temp_file = tmp_path / "temp_file.txt"
+        temp_file.touch()
+
+        with patch.object(
+            S3AssetManager, "prepare_paths_for_upload"
+        ) as prepare_paths_mock, patch.object(
+            api, "hash_attachments", return_value=(Mock(), [])
+        ) as hash_attachments_mock:
+
+            asset_group_mock = AssetRootGroup(
+                root_path=root_dir,
+                inputs=set(),
+                outputs=set(),
+                references=set(),
+            )
+            upload_group_mock = Mock(
+                asset_groups=[asset_group_mock],
+                total_input_files=1,
+                total_input_bytes=100,
+            )
+            prepare_paths_mock.return_value = upload_group_mock
+
+            runner = CliRunner()
+            result = runner.invoke(
+                main,
+                [
+                    "asset",
+                    "snapshot",
+                    "--root-dir",
+                    root_dir,
+                    "--manifest-out",
+                    str(manifest_out_dir),
+                ],
+            )
+
+            assert result.exit_code == 0
+            prepare_paths_mock.assert_called_once_with(
+                input_paths=[str(temp_file)], output_paths=[root_dir], referenced_paths=[]
+            )
+            hash_attachments_mock.assert_called_once()
+
+    def test_invalid_manifest_out(self, tmp_path):
+        """
+        Tests if CLI snapshot raises error when called with invalid --manifest-out
+        """
+        root_dir = str(tmp_path)
+        invalid_manifest_out = str(tmp_path / "nonexistent_dir")
+
+        runner = CliRunner()
+        result = runner.invoke(
+            main,
+            ["asset", "snapshot", "--root-dir", root_dir, "--manifest-out", invalid_manifest_out],
+        )
+
+        assert result.exit_code != 0
+        assert (
+            f"Specified destination directory {invalid_manifest_out} does not exist. "
+            in result.output
+        )
+
+    def test_asset_snapshot_recursive(self, tmp_path):
+        """
+        Tests if CLI snapshot --recursive flag is called correctly
+        """
+        root_dir = str(tmp_path)
+        subdir1 = tmp_path / "subdir1"
+        subdir2 = tmp_path / "subdir2"
+        subdir1.mkdir()
+        subdir2.mkdir()
+        (subdir1 / "file1.txt").touch()
+        (subdir2 / "file2.txt").touch()
+
+        with patch.object(
+            S3AssetManager, "prepare_paths_for_upload"
+        ) as prepare_paths_mock, patch.object(
+            api, "hash_attachments", return_value=(Mock(), [])
+        ) as hash_attachments_mock:
```

Comment:
I see we've applied both of these mocks three times. Should we consider using [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html) for these mocks?